store/sqlite.go

```diff
@@ -218,145 +224,97 @@ func (db *DB) pgSeedFile(name string) error {
 
 // migrate sets up migration tracking and executes pending migration files.
 //
-// Migration files are embedded in the sqlite/migration folder and are executed
+// Migration files are embedded in the migration folder and are executed
 // in lexicographical order.
 //
-// We prepend each migration file with PRAGMA user_version = xxx; Each migration
+// We prepend each migration file with version = xxx; Each migration
 // file run in a transaction to prevent partial migrations.
-func (db *DB) migrate() error {
+func (db *DB) migrate(ctx context.Context, d dbdriver.Driver, curVer version, databaseName string) error {
 	db.l.Info("Apply database migration if needed...")
-
-	curVer, err := db.version()
-	if err != nil {
-		return fmt.Errorf("failed to get current schema version: %w", err)
-	}
-
 	db.l.Info(fmt.Sprintf("Current schema version before migration: %s", curVer))
 
 	// major version is 0 when the store isn't yet setup for the first time.
 	if curVer.major != 0 && curVer.major != majorSchemaVervion {
 		return fmt.Errorf("current major schema version %d is different from the major schema version %d this release %s expects", curVer.major, majorSchemaVervion, db.releaseVersion)
 	}
 
-	// Apply migrations
-	names, err := fs.Glob(migrationFS, "migration/*.sql")
-	if err != nil {
-		return err
+	if curVer.major == 0 && curVer.minor == 0 {
+		createDatabaseStatement := fmt.Sprintf("CREATE DATABASE %s", databaseName)
+		if _, _, err := d.ExecuteMigration(
+			ctx,
+			&dbdriver.MigrationInfo{
+				ReleaseVersion: db.releaseVersion,
+				Version:        createDatabaseSchemaVersion,
+				Namespace:      databaseName,
+				Database:       databaseName,
+				Environment:    "bb.environment.internal",
+				Engine:         dbdriver.UI,
```

Comment:
Adding a new engine type: LIBRARY?

Also I am wondering if we should rename engine to source because UI, VCS, LIBRARY all use the same engine but from different sources. And we can also say flyway is a different source.

Reply:
Done.

Make sense, I'll make the renaming in a follow-up CL.